### ISSUE TITLE
IMPORTANT: Bot warning with this bot. Red message in app

### ISSUE THREAD

**Issue by goedzo** [PokemonGoF/PokemonGo-Bot#6198]
IMPORTANT! It seems that almost all of our bot accounts got flagged with the red warning. So stop using the bot. We must investigate the cause.

I suspect that it is caused by the inbox is not cleared. When i open app, i get very old messages being displayed. I suspect this causes the bot warning now. We need to fix this asap!

On my account that i use daily on my phone in combination with the app, no red warning is there.

**Comment by MerlionRock** [comment]
@Davidakachaos take a look when you are able to. This bot not receiving inbox messages the way it should.

I'll also take a look later.

**Comment by VoIP** [comment]
I am able to replicate this issue. As another update, if its not wiped after a day, it triggers a niantic warning.

**Comment by MerlionRock** [comment]
Temporary solution for those who are still safe: login phone to clear inbox, disable gym slot

**Comment by MerlionRock** [comment]
Line 1532, change request.get_inbox() to request.get_inbox(history=true)

I need someone with multiple accounts that hasn't got the warning or didn't bot since we update to version 0.69.1 to test this. One acct with the above change, one without and let us know what happen to them

**Comment by MerlionRock** [comment]
I don't have time, i'll paste the response here and hopefully someone can figure this out asap
```
[{'notification_id': u'GYM_REMOVAL_aa47efe6d5004151b77e16f6bb0368e2.16', 'category': u'gym_removal', 'variables': [{'literal': u'aa47efe6d5004151b77e16f6bb0368e2.16', 'name': u'GYM_ID'}, {'literal': u'Shangri la Phoenix', 'name': u'GYM_NAME'}, {'literal': u'10901816661029224455', 'name': u'POKEMON_ID'}, {'lookup_table': u'pokemon', 'name': u'POKEDEX_ENTRY_NUMBER', 'key': u'131'}, {'name': u'POKEMON_DISPLAY', 'byte_value': '\x10\x02'}, {'literal': u'GYM_BADGE_VANILLA', 'name': u'GYM_BADGE_RANK'}, {'literal': u'http://lh5.ggpht.com/HOLIHu6SKCDB45sW4e-Y0SLPO4owOk4RK27xirKax3O6AK9WcI9ltWJcosDA-CY6Y2sTWXAsB0lxZKBtIICq', 'name': u'GYM_BADGE_IMAGE_URL'}, {'literal': u'false', 'name': u'GYM_BADGE_LEVEL_UP'}, {'literal': u'127', 'name': u'GYM_BADGE_EARNED_POINTS'}, {'literal': u'0.454', 'name': u'GYM_BADGE_PROGRESS'}, {'literal': u'12', 'name': u'POKECOIN_AWARDED'}, {'literal': u'0', 'name': u'POKECOIN_AWARDED_TODAY'}, {'literal': u'50', 'name': u'MAX_DAILY_POKECOIN'}, {'literal': u'0', 'name': u'BATTLES_WON'}, {'literal': u'3', 'name': u'BATTLES_LOST'}, {'literal': u'0', 'name': u'DEPLOYED_MILLIS'}], 'labels': [1, 3], 'create_timestamp_ms': 1500532796709L, 'title_key': u'gym_removal_notification_toast'}, 
```

label
UNREAD = 1;
NEW = 2;
IMMEDIATE = 3;

Server will always return as UNREAD. Read to figure out how to inform the server that message is read. It might be something new in 0.69.1

**Comment by VoIP** [comment]
As a better solution, might be better to disable gym until its resolved, and pass it along to Dev Repo.

**Comment by MerlionRock** [comment]
@VoIP likely not gym or inbox issue. Mappers who don't gym are hit too.

Likely to be 0.69.1 api issue.

**Comment by arbosus** [comment]
Could it be with the buddy pokemon? I will log on and get Magikarp found a candy +26 magikarp candies

**Comment by MerlionRock** [comment]
@arbosus no, mappers don't do any of those. They are being hit too

**Comment by camnomis** [comment]
Today sucks!

Was playing around with optimiser as transfer wasn't working on my usual config, something went wrong and lost a ton of good mons (Unown & Moltres included) AND got the warning message, where is that cry smiley!

**Comment by goedzo** [comment]
Well my phone account also received the warning. So it is not caused by inbox!

**Comment by thelemonenergy** [comment]
I don't know if this is helpful, but my gf who stopped botting 2 weeks ago did not receive a warning.
I went on till i read this right now, checked it and I did get the warning.

**Comment by goedzo** [comment]
@thelemonenergy, so your girlfriend is safe since she did not bot two weeks ago? If that is the case, definately something has changed in 0.69.1 that allows detection. Did she used gym slotting two weeks ago?

**Comment by thelemonenergy** [comment]
She did not use any gym features. (Luckily I configured her bot so i know most of the config)
I think the 0.69.1 was released earlier than 2 weeks ago, but the gym slotting was not in the master branch by then.

**Comment by MerlionRock** [comment]
@thelemonenergy our not only updated to 0.69.1 about 2-3 days ago.

**Comment by MerlionRock** [comment]
Guys, stay clam. This is not the first time (and definitely won't be the last). When. This happen stay clam, keep playing (as in please be good boy and girl, stop bot) until warning disappear. By then, we should have figured out what causes the flag and we will be back botting in no time. :)

**Comment by thelemonenergy** [comment]
Oh, ok sorry, i kind of mixed that up. I will look in the event file of the webinterface and tell you all i can see about the date and version. Will do as soon as I have access to her pc.

**Comment by VoIP** [comment]
Appears its a common issue in the Dev Pogo spot. Most likely there is a hidden proto or something that causes the issues.

**Comment by MerlionRock** [comment]
@VoIP yup this is the 2nd time it happened.... The last time it happened, it takes almost a month before they say, ya, if you don't want to see the warning, please do this and that ... Hopefully this time round they will be faster....

**Comment by VoIP** [comment]
@MerlionRock  and possibly hope we get a "compensated" for the issues.. but I highly doubt that will happen (even if you are doing mapping).

**Comment by camnomis** [comment]
Any update on this yet?

Is #re-research on discord the best place to follow for updates, tried to keep up but get lost quite quickly

**Comment by VoIP** [comment]
@camnomis still not stable. #re-research is still looking into it, but there **might** be a fix in one of the node.js api's.

**Comment by goedzo** [comment]
We've implemented the fix from node.js - Please test this by using this API, and report your results:
https://github.com/goedzo/pgoapi/tree/Test-Anti-Ban

**Comment by camnomis** [comment]
Using your fix @goedzo and so far so good (spun up a brand new account to test it on)

**Comment by goedzo** [comment]
Ok cool @camnomis keep us posted!

**Comment by nbq** [comment]
Im running it too with no errors on the account so far so for me it's a pass

**Comment by arbosus** [comment]
as soon as i got this warning i created a new account and put it on the bot it took 4 days for the warning to show up.

**Comment by goedzo** [comment]
Lets make sure it's ok by running it for at least 4 days, to be sure.

**Comment by nbq** [comment]
yep, testing it for a few days never hurts anyone

**Comment by camnomis** [comment]
Still no warning

**Comment by MerlionRock** [comment]
Would advise not to run bot on accounts that matter.

The latest batch is still not safe.

**Comment by camnomis** [comment]
I am using a throw away account created for testing of @goedzo's fix, I would reiterate @MerlionRock and advise against using on an account you care about

**Comment by Jcolomar** [comment]
[2017-08-16 14:04:29] [PokemonGoBot] [INFO] Probably permabanned, Game Over ! Play again at https://club.pokemon.com/us/pokemon-trainer-club/sign-up/

**Comment by camnomis** [comment]
@Jcolomar is that on a new account with the fix?

**Comment by Jcolomar** [comment]
its and old one
the new one is shadowbanned

**Comment by Jcolomar** [comment]
https://www.reddit.com/r/pokemongodev/comments/6cng1n/current_guess_at_shadowbanned_pokemon/

**Comment by Jcolomar** [comment]
no good, evolve pokemons in 25 min, its weird

**Comment by MerlionRock** [comment]
Guys, stop botting, incoming ban wave.

**Comment by Jcolomar** [comment]
My both account was banned, rip

**Comment by MerlionRock** [comment]
Your level 40 too?

**Comment by Jcolomar** [comment]
38 and 39

**Comment by MerlionRock** [comment]
Straight ban, or already have warning, follow by ban?

**Comment by Jcolomar** [comment]
i was warned

**Comment by camnomis** [comment]
This is the state of my various accounts:

Bot account - Lasted botted 4th Aug, no warning message
Main account - Lasted botted "pre fix" 6th August, ran once 13th aug post "fix", warning message till apprearing
New bot - Created 13th Aug and only ran with @goedzo's "fix" no warning message

**Comment by camnomis** [comment]
Ok, warning message has disappeared from main account, but looks like it has a shadow ban (Unown spawned but didn't show up on my account), also the "Slash of Shame" has appeared on a few on of the premier mons I have used the bot to catch on my main account.

Newly spun up account is still working fine with nor warnings.

**Comment by Magicode1** [comment]
I am very new to this botting scene. Made an account yesterday evening and it ran fine until it's sleep schedule this morning. How would one install the pgoapi fix that @goedzo is working on?

**Comment by MerlionRock** [comment]
@Magicode, if you just 8nstalled, it's already running g the "fix", however, the fix is not working.

Actually, currently NO BOT is safe. If any bot claim that they are safe, they are just after your $$.

Strongly advise to wait until we post a notice to go ahead. As of now, you might just want to create a new account just to see what the bot can do. You can also join us in discord for discussion.

**Comment by Magicode1** [comment]
Alright, sounds good. This bot is running a new account, just to see. I will Keep you updated on what happens, but it hasn't been flagged yet.

**Comment by MerlionRock** [comment]
Wait 3-7 days. Of you are lucky, it's just warning and Softban, if you are unlucky. It's a straight ban.

**Comment by camnomis** [comment]
Was bored so logged into my test account on my phone, considering EVERYTHING on this account was bot caught or hatched almost 50% of the mons had the “Slash of Shame” so still no where near safe to use on an account you care about

**Comment by MerlionRock** [comment]
@camnomis actually, you have a very interesting finding. IF Niantic detect us of cheating is because of a faulty API, then logically, all mons should be marked as slash.

Perhaps you want to help me to see if there's a pattern between those marked and those without? Maybe time and date they are caught? Or type of mons?

**Comment by camnomis** [comment]
I did a mass delete, 

I’ll turn off egg hatching and not delete any next time I’ve build up a lot and do a compare

**Comment by MerlionRock** [comment]
@camnomis I think you better stop botting temporary. I have just coded in killswitch to dev branch.

Of cos, you can overwrite the killswitch if you want to.

**Comment by Magicode1** [comment]
Alright, here is the roundup from the test I ran on some completely fresh
accounts.

1. I ran the bot for 20 hours straight, and nothing was flagged, and no
message appeared. I did not even get a single "Slash of Shame" on any of my
pokemon. The not went from level 1 to 19 in that time, and it was set up in
my home town. I had 'Complete Tutorial' on and it was putting pokemon in
Gyms, spinning pokestops, evolving, etc.

2. My second account was set up in Times Square, New York, and ran for
about 2 days with the default sleep schedule. It went from level 1 to 21 in
that time and, again, nothing appeared, with no "Slashes of Shame". Was
doing the same thing as the first bot in terms of activities.

3. Another bot in my hometown, this on went from 1 to 9 because I only ran
it for about 5 hours. Was not putting pokemon in Gyms, but doing everything
else. One thing I did notice on this account though, was that when the bot
failed to set a nickname on "complete tutorial", it just started catching
pokemon, and that got me IP banned from that VPS. On a different device
though, it continued to work just fine, with no warnings or
"Slashes of Shame".

TL;DR

None of my accounts where flagged, minus a small bug in the "Complete
Tutorial", and no slashes of shame appeared. I will continue to run bots
and update you on my findings.

**Comment by MerlionRock** [comment]
@Magicode1, one or two more days you should see Softban, follow by warning~ and eventually ban.

**Comment by camnomis** [comment]
Game over, thanks for playing please try again!

Took about 6 days from spinning up a brand new account to getting permabanned

**Comment by Jcolomar** [comment]
i started new acc. Only on phone, sniping only in philadelphia city spoofing gps. Lets see what happend in some days

**Comment by highcmd** [comment]
I had warning message and now for a 4-5 days im still having "failed to get game data from server" and on ptc it says that i have 1 lvl no pokemons etc, is this some kind of ban?

**Comment by thelemonenergy** [comment]
That looks like a perma ban.

Btw my red message disappeared now after 2 weeks of not botting.

**Comment by Magicode1** [comment]
Still haven't gotten anything on my bots except one (Note to self, don't
teleport your not randomly across the country to Anaheim to get Kangaskan
:b)

**Comment by LazeBear** [comment]
any updates? Is the bot safe now?

**Comment by Freeman19** [comment]
have any update ???

**Comment by highcmd** [comment]
its working well also i got back my account, after 2 weeks i tried to login and everything was like before and without warning message

**Comment by MonsterMap** [comment]
Still not safe. All my accounts were banned in about 4 days.

**Comment by nbq** [comment]
I think you will find that with pretty much any bot right now. There is still a lot we have yet to figure out (as a community) with Niantic cracking down.

**Comment by MerlionRock** [comment]
@MonsterMap Those accounts will be back in a few weeks. That's how the mapping community rotate their accounts.
